### PR TITLE
Use VecDeque for dropping the most previous metrics if it's more than 6 hours ago

### PR DESCRIPTION
**Pull request by Krout0n**
Issue https://github.com/mackerelio-labs/mackerel-agent-rs/pull/11#discussion_r517778010
It was pointed out that unlimited size of metrics isn't a good way. So, I made Agent drop previous metrics when its size is reaching out the limit.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,5 +1,6 @@
 use metric::{HostMetric, MetricValue};
 use std::{
+    collections::VecDeque,
```
Comment (ne-sachirou) [inline src/lib.rs:3]: memo:
> A double-ended queue implemented with a growable ring buffer.
The "default" usage of this type as a queue is to use push_back to add to the queue, and pop_front to remove from the queue. extend and append push onto the back in this manner, and iterating over VecDeque goes front to back.

File: src/lib.rs [hunk 2]
```diff
@@ -96,12 +97,28 @@ impl Agent {
...
+        let result = self.client.post_metrics(post_metrics_with_heaped).await;
+
         // If Ok, then heaped metric must be empty, else extend it.
-        // TODO: Drop metric if it is too ancient.
-        self.metric = if result.is_ok() { vec![] } else { metric };
+        self.heaped_metrics = if result.is_ok() {
+            VecDeque::new()
+        } else {
+            // Drop the most previous metrics if it's more than 6 hours ago.
+            if 60 * 6 <= self.heaped_metrics.len() {
```
Comment (ne-sachirou) [inline src/lib.rs]: Plaese name this value `60 * 6`. e.g. `max_heap_size`.

Comment (ne-sachirou) [reply src/lib.rs]: memo: This is the same value as mackerel-agent Go.

```go
	postMetricsBufferSize          = 6 * 60 // Keep metric values of 6 hours in the queue
```

File: src/lib.rs [hunk 3]
```diff
@@ -110,10 +112,10 @@ impl Agent {
 
         // If Ok, then heaped metric must be empty, else extend it.
         self.heaped_metrics = if result.is_ok() {
-            VecDeque::new()
+            VecDeque::with_capacity(MAX_HEAP_SIZE)
```
Comment (ne-sachirou) [inline src/lib.rs:115]: 👍

**Review by ne-sachirou — APPROVED** [review]
🙆‍♀️

**Comment by Krout0n** [comment]
Thank you!

**Pull request merged**